cellfinder_napari/curation.py

```diff
@@ -513,59 +516,74 @@ def __save_yaml_file(self):
         yaml_contents = {"data": yaml_section}
         save_yaml(yaml_contents, yaml_filename)
 
+    def update_progress(self, attributes: dict):
+        """
+        Update progress bar with ``attributes``.
+        """
+        for attr in attributes:
+            self.progress_bar.__setattr__(attr, attributes[attr])
 
-@thread_worker
-def extract_cubes(
-    cells_to_extract: List[Cell],
-    non_cells_to_extract: List[Cell],
-    output_directory: Path,
-    signal_array: np.ndarray,
-    background_array: np.ndarray,
-    voxel_sizes: Tuple[int, int, int],
-    network_voxel_sizes: Tuple[int, int, int],
-    batch_size: int,
-    cube_width: int,
-    cube_height: int,
-    cube_depth: int,
-):
-    from cellfinder_core.classify.cube_generator import CubeGeneratorFromFile
-
-    to_extract = {
-        "cells": cells_to_extract,
-        "non_cells": non_cells_to_extract,
-    }
-
-    for cell_type, cell_list in to_extract.items():
-        print(f"Extracting type: {cell_type}")
-        cell_type_output_directory = output_directory / cell_type
-        print(f"Saving to: {cell_type_output_directory}")
-        ensure_directory_exists(str(cell_type_output_directory))
-
-        cube_generator = CubeGeneratorFromFile(
-            cell_list,
-            signal_array,
-            background_array,
-            voxel_sizes,
-            network_voxel_sizes,
-            batch_size=batch_size,
-            cube_width=cube_width,
-            cube_height=cube_height,
-            cube_depth=cube_depth,
-            extract=True,
-        )
+    def update_status_label(self, label: str):
+        self.status_label.setText(label)
 
-        extract_batches(cube_generator, cell_type_output_directory)
-        print("Done")
+    def extract_cubes(self):
+        """
+        Yields
+        ------
+        dict
+            Attributes used to update a progress bar. The keys can be any of
+            the properties of `magicgui.widgets.ProgressBar`.
+        """
+        from cellfinder_core.classify.cube_generator import (
+            CubeGeneratorFromFile,
+        )
 
+        to_extract = {
+            "cells": self.cells_to_extract,
+            "non_cells": self.non_cells_to_extract,
+        }
+
+        for cell_type, cell_list in to_extract.items():
+            cell_type_output_directory = self.output_directory / cell_type
+            ensure_directory_exists(str(cell_type_output_directory))
```

Comment:
Assume a `Path` won't work here?

Reply:
[I think a `Path` object should work](https://github.com/adamltyson/imlib/blob/625193be4a586d9040a48df9d51dbdd3a17c7d06/imlib/general/system.py#L46-L56). More generally I think the `imlib` dependency should go, but that's a problem for another time.

Reply:
I switched to just using `Path.mkdir()` here.